2_segmenting_data/README.md

```diff
@@ -0,0 +1,104 @@
+Since CellPose struggled with segmenting cytoplasm using only the RFP channel, I use a function to overlay nuclei, ER, and cytoplasm channels into one image from each well and site called `overlay_channels()` in [segmentation_utils.py](segmentation_utils.py).
+ER channel is not used by CellPose during segmentation as CellPose can only use 2 channels, one as a base and one to segment.
+
+**Note:** The channel numbers correspond to the CellPose channel colors.
+The `overlay_channels()` function makes cytoplasm the red channel of the image and nuclei the blue channel of the image.
+Thus, `channels : [1,3]` has CellPose segment the cytoplasm channel (3: red) using the nuclei channel (1: blue) as its base.
+
+- `diameter : 155` This parameter sets the cell diameter of the cells within the image to a value, which can be calculated using the `calculate` button. 
+I found that the segmentation worked better when the diameter was established with a set value.
+- `flow_threshold : 0.4` This paramenter decreases the maximum allowed error of the flows for each mask (default is `flow_threshold : 0.4`).
+- `remove_edge_masks : True` This parameter removes any masks from CellPose that are touching an edge of the image.
+
+To show how I determined the best parameters for cytoplasm segmentation, below is a figure showing a cropped image of cells from the cytoplasm channel (**panel A**) and various cytoplasm segmentation methods. 
+
+![Cytoplasm_Segmented](Example_Figures/Cytoplasm_Segmentation_Fig.png)
+
+In **panel B**, it shows how CellPose does with only segmenting cytoplasm using the greyscale image from the RFP channel.
+Since it has no indication of what is a cell, it struggles with any parameters for properly segment cytoplasm, which this image is an example of one set of parameters.
+
+In **panel C**, this shows how CellPose segments the cytoplasm with non-optimal parameters and with the channels overlayed into a single image. 
+As seen with the cell with the orange filter, these parameters oversegments the cell and does not keep the shape of that cell. 
+As well, with the green filtered cell to the left of the orange cell, CellPose undersegments this cell and does not include a lot of the cytoplasm within the outlines. 
+
+In **panel D**, this image shows the parameters I used to segment the cytoplasm.
+These parameters worked the best for all images and it did the best in segmenting each cell without over or undersegmenting. 
+
+## Linking Cells with Cell IDs
+
+In this pipeline, after finding the nuclei center coordinates and the cytoplasm outlines, the next step is to determine which nuclei and cytoplasms are related.
+This will be helpful during the analysis steps later in this project.
+
+First, a `Cell_ID` to is assigned to every cytoplasm using the `index` column.
+
+Then, using [matplotlib.path](https://matplotlib.org/stable/api/path_api.html), this function forms a polygon based on the cytoplasm outlines.
```

Comment:
This is really cool. Innovative!

Can you comment on what happens if more than one nucleus are observed in the same polygon?

Reply:
Sure! It is what Roshan and I did for Cell-Health. Essentially the nucleus from the same cytoplasm (polygon) will have the same cell ID.